poleno/utils/static/poleno/js/range_widget.js

```diff
@@ -0,0 +1,14 @@
+/* Required for ``poleno.utils.forms.RangeWidget`` form widget.
+ */
+
+$(function(){
+	$('.pln-range-widget').each(function(){
+		var slider = $(this).children('input').eq(0);
+		var output = $(this).children('span').eq(0);
+		output.html($(slider).val());
+           $(slider).on('input', valueUpdated);
```

Comment:
Idealne vzdy pisat taketo "rozsirenia" funkcionality DOM elementov tak, aby fugovali aj ked sa DOM element prida do dokumentu neskor ajaxom.

Preto namiesto selectnutia elementov na zaciatku a zapnutia event handlera iba pre selectnute elementy, je lepsie zavesit listener dynamicky. Nieco taketo nefunguje?

```
$(document).on('input', '.pln-range-widget input', function(event) {
  ...
})
```

Ak si spravne pamatam, to by mali pocivat na input eventy, na **vsetkych** elementoch, aj tych, co sa pridaju do DOM neskor cez ajax.

Vid napriklad ako je spravene `autocomplete.js` alebo `composite_text.js` alebo dalsie

Reply:
ano funguje, vdaka

Reply:
@viliambalaz Pozor, uvodne `output.html($(slider).val());` mas iba v toplevel `$(function(){...})`, ktore sa spusti iba na zaciatku po naloadovani dokumentu. Takze ak `RangeWidget` vlozis na web neskor ajaxom, tak sa uvodne `output.html($(slider).val());` na nom nevykona a ostane bez cisla, kym sa prvy krat neemitne event `input`.

Takze bud potrebujes cislo vyrendrovat uz v template, aby uvodne `output.html($(slider).val());` nebolo treba vobec, alebo sa potrebujes zavesit na eventy zmeny DOM a pridanych `RangeWidget` elementoch to cislo doplnit. Ajax je implementovany tak, ze pri zmene DOM posiela `pln-dom-changed` event, vid `ajax.js`. Pozri ako sa ten event pouziva napriklad v ostatnych skriptoch v `static/poleno/js/` adresari.

Kazdopadne, jednoduchsie byde asi uvodne cislo vyrendrovat priamo v template.

Reply:
@martinmacko47 Ano, najprv som skusal hodnotu vyrendrovat priamo v template, ale nastal problem v tejto situacii. Po nacitani stranky som posunul sliderom na inu hodnotu a v prehliadaci som si obnovil stranku (`F5`). Slider ostal na nezmenenej polohe a hodnota v `span` sa zmenila, nacital sa udaj z databazy, takze nastala nezhoda medzi sliderom a `span`. Preto som explicitne nastavil hodnotu v `span` `output.html($(slider).val());`. Takze by som to nechal tam.

Aky bude sposob riesenia?
1) Pouzili by sme kombinaciu sposobov  inicializacie `output.html($(slider).val());` a sucasne rendrovania hodnoty v template.

2) Pomocou `pln-dom-changed` eventu. 
P.S. Ako si mozem odsimulovat ,aby som  `pln-dom-changed` event zavolal?

Reply:
@viliambalaz Ano, v takom pripade treba pouzit `output.html($(slider).val());`, nech mame aktualnu hodnotu slidera. Takze bude sa treba zavesit na `pln-dom-changed` a hodnotu updatnut aj vtedy. Analogicky ako pri inych js veciach, co tiez nieco potrebuju spravit, ked sa element prida do DOM.

`pln-dom-changed` si mozes odsimulovat tak, ze widget so sliderom vlozis do nejakeho formulara, ktory sa nacitava ajaxom. Cez ajax sa loaduje napriklad obsah snooze popupu:

![snooze1](https://user-images.githubusercontent.com/5547275/81667778-fefdff80-9443-11ea-84bf-fb2d6ca8849d.png)

![snooze2](https://user-images.githubusercontent.com/5547275/81667782-01f8f000-9444-11ea-968f-5dc02ee1686a.png)

Snooze button sa pri akcii zobrazi, ak akcia ma obligee deadline, ktory v priebehu poslednych par dni (tusim 3) expiroval. V popupe sa zobrazi `SnoozeForm`. Ak do neho pridas fieldy, tak by sa mali zobrazit v popupe. Mozno budes potrebovat upravit template pre ten popup.

Reply:
:ok: